torch/testing/_internal/distributed/pipe_with_ddp_test.py

```diff
@@ -0,0 +1,101 @@
+from torch.nn.parallel import DistributedDataParallel
+import torch.distributed as dist
+from torch import nn
+import torch
+from torch.testing._internal.dist_utils import INIT_METHOD_TEMPLATE, dist_init
+from torch.testing._internal.common_distributed import (
+    requires_gloo,
+    requires_nccl,
+    skip_if_lt_x_gpu,
+    skip_if_rocm,
+)
+from torch.testing._internal.distributed.rpc.rpc_agent_test_fixture import (
+    RpcAgentTestFixture,
+)
+from torch.distributed._pipeline.sync import Pipe
+import unittest
+
+class PipeWithDDPTest(RpcAgentTestFixture):
+    @property
+    def world_size(self) -> int:
+        return 4
+
+    @skip_if_lt_x_gpu(4)
```

Comment:
IIRC, we will need to run ci-all to trigger multigpu tests to cover this

Reply:
Tests are passing for multigpu: https://app.circleci.com/pipelines/github/pytorch/pytorch/250277/workflows/77bc3b3b-2a7c-46f7-b0ea-e06a496d1e4b/jobs/9579475, PR: https://github.com/pytorch/pytorch/pull/49302